### ISSUE TITLE
DNS server of slirp4netns only forwards to primary DNS server of Host system

### ISSUE THREAD

**Issue by cpp977** [rootless-containers/slirp4netns#347]
I am facing an issue when using slirp4netns with rootless podman.

A DNS server with address `10.0.2.3` is added to the `/etc/resolv.conf` of the rootless Container at primary position.
This server is followed up by the nameservers of the host's `/etc/resolv.conf`.
I could not find any documentation about how this DNS server is implemented but I experienced the following behavior:
When asking inside the container for a hostname address outside the host, the `10.0.2.3`-Server forwards the query to the primary DNS Server of the host's `/etc/resolv.conf`. 
If this is not available or cannot answer the query the `10.0.2.3` server returns that he cannot answer the query.
Afterwards, the next nameserver of the container's `/etc/resolv.conf` is queried which is actually the same nameserver as before. So this query will fail as well. Only then a different nameserver is asked (the second from the host's `/etc/resolv.conf`).

I would expect, that either the `10.0.2.3` server forwards to all nameservers from the host's `/etc/resolv.conf` in order or does not forward at all but return that he cannot answer the query.

TLDR
Can you point me to documentation (or to a place in the implementation) of the slirp4netns DNS server?

**Comment by AkihiroSuda** [comment]
> a place in the implementation

Here:
https://gitlab.freedesktop.org/slirp/libslirp/-/blob/v4.8.0/src/slirp.c?ref_type=tags#L434

**Comment by cpp977** [comment]
Thanks, I will have a look after the summer holidays.

**Comment by cpp977** [comment]
This issue explains the observed behavior: https://gitlab.freedesktop.org/slirp/libslirp/-/issues/26

This is therefor not relevant here, but has to be fixed in slirp.
